nestkernel/nest_types.h

```diff
@@ -145,62 +141,23 @@ const synindex invalid_synindex = MAX_SYN_ID;
 //! target index into thread local node vector
 typedef unsigned short targetindex;
```

Comment:
Shouldn't we get rid of all typedefs here? There are 3 left which we haven't discussed yet:  
`synindex`: Currently `unsigned int`, but only gets 9 Bits when sent over MPI. Primarily used as index, should therefore become `size_t` as all places where memory matters bitfields are used anyway.  
`targetindex`: Currently `unsigned short` and not sent over MPI. Here, size actually matters and should therefore either keep `unsigned short` or could be changed to `size_t` together with a bitfield (16b) in the targetidentifier. The first would require an implicit conversion from `unsigned short` to `size_t` while the latter would require some bitmasking and zero-padding to get the actual `size_t` value. Optimized compiled code will probably do the exact same for both, so I think using `size_t` with a bitfield ist actually cleaner here to indicate that this is indeed an index.  
`tic_t`: Unsure about this one, this might actually be a valid typedef.

Reply:
My take on this was that as these `typedef`s are actually descriptive names for something and thus they could stay as is. Also, I thought, if we want to change their definition, it would be easier if they were defined at a central point. If there is no danger to increase memory requirements by changing them, I might reconsider.

@suku248:  What is your opinion on this?

Reply:
The `targetindex` is only ever used in a single file (*target_identifier.h*), so I think instead of a typedef at a central point, one might just change the type directly in that file (or use a local typedef there). Of course, having it at a central point raised the awareness to that typedef and somewhat defines it as a parameter which could be changed if desired. However, changing the type without thinking about it carefully might increase the memory usage much more than expected (due to alignment).  
For the `synindex` we already have the `NUM_BITS_SYN_ID` which controls the size at all places where size actually matters.  
So my conclusion would be that it is safe to remove these typedefs. But I am quite interested in Susanne's opinion as well.

Reply:
`TargetIdentifierIndex::target_` must be 16 bit and not longer to keep HPC-Synapse small. Changing the type to `size_t` would increase memory requirement for these synapses by at least 48 bit (see https://github.com/nest/nest-simulator/blob/4192435db972ce333e6a33e5769407fd9d8c3b06/nestkernel/connection.h#L308). So the underlying type must remain `unsigned short` and I would argue to use that explicitly instead of `typedef'ing`.

Reply:
Actually you are right, thought about it again and we musn't chang ethe type. Using 32b would be fine, as we have 16b offset at the moment anyway (as the SynIdDelay forces a 32b alignment), but 64b (size_t) would increase the memory footprint.  

```cpp
typedef unsigned short targetindex;
targetindex target;
```
will not be the same as
```cpp
size_t target : 16;
```
which is what I first thought. But actually, the bitfields will always allocate the size of the largest member type (i.e., size_t), even if only using 16b of that type. The TargetIdentifierIndex will thus have a size of 8B instead of the 2B it currently has. Thanks Hans Ekkehard!  

Still, then we should remove the typedef and use the `unsigned short` explicitly as you suggested.